components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,294 @@
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.2 Requirement: Compute and store the $z$ location of a given level
+
+The $z$ location at the top of each level will be computed and stored within the vertical coordinate module.
+In the model, $z$ is defined as being positive up from the geoid.
+The $z$ location is calculated in a bottom-up summation of the mass thickness times the specific volume, starting with the known bottom elevation.
+The $z$ location of the interfaces is needed to compute the geopotential gradient and the sea level.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.3 Requirement: Compute and store the vertical level bounds for each horizontal cell
+
+All tendency terms require a variable vertical loop range to account for variation in bottom elevation and ice shelf cavities.
+These bounds will be computed in the vertical coordinate module and used for setting masking arrays that will be used to determine where calculations are valid in the vertical.
+
+### 2.4 Requirement: Compute and store the geopotential
+
+The geopotential will be computed and stored within the vertical coordinate module.
+Initially this will be calculated by summing together the $z$ location times $g$.
+In the future, tidal potential and self attraction and loading will also have contributions to the geopotential.
+
+### 2.5 Requirement: Compute and store the desired thickness used in the $p^\star$ vertical coordinate
+
+The $p^\star$ vertical coordinate specifies a desired layer thickness that stretches the initial thicknesses by the bottom pressure anomaly.
+This involves adding a vertically-weighted bottom pressure perturbation to the reference (initial) thickness.
+This desired thickness will be computed and stored within the vertical coordinate module.
+The vertical advection algorithm will use this desired thickness to compute the vertical transport.
+
+### Desired: General Arbitrary Lagrangian Eulerian (ALE) coordinate support
+
+In future versions of Omega, the $p^\star$ coordinate will be extended to a more general ALE coordinate
+
+### Desired: Vertical Lagrangian remapping (VLR) support
+
+The vertical coordinate module should retain flexibility to support VLR in future versions of Omega.
+
+## 3 Algorithmic Formulation
+The non-Boussinesq momentum equation is
+
+$$ 
+\frac{D \mathbf{u}_h}{D t } + f\boldsymbol{k}\times \mathbf{u}_h + \left(\alpha\nabla_A p + \nabla_A \Phi \right) = \boldsymbol{\mathcal{F}},
+$$
+
+where $\mathbf{u}_h$ is the horizontal velocity, $f$ is the Coriolis parameter, $\alpha = \frac{1}{\rho}$ is the specific volume, $\rho = \rho(\Theta,S,p)$ is the density, $p$ is the hydrostatic pressure, $\Phi$ is the geopotential, and $\boldsymbol{\mathcal{F}}$ are the dissipative terms.
```

Comment:
Alpha is not rendering correctly here, maybe try this?
```suggestion
where $\mathbf{u}_h$ is the horizontal velocity, $f$ is the Coriolis parameter, $\alpha$ = $\frac{1}{\rho}$ is the specific volume, $\rho = \rho(\Theta,S,p)$ is the density, $p$ is the hydrostatic pressure, $\Phi$ is the geopotential, and $\boldsymbol{\mathcal{F}}$ are the dissipative terms.
```

Reply:
good catch!